Runtime/Properties/SnapZoneProperty.cs

```diff
@@ -62,23 +70,90 @@ public SnapZone SnapZone
         }
 
         private SnapZone snapZone;
-        
+
+        protected virtual void Reset()
+        {
+            if (SnapZone.attachTransform)
+            {
+                return;
+            }
+
+            GameObject attach = new GameObject("Attach Point");
+            attach.transform.SetParent(transform, false);
+            attach.transform.localRotation = Quaternion.identity;
+            attach.transform.localPosition = Vector3.zero;
+            attach.transform.localScale = Vector3.one;
+
+            FieldInfo attachField = typeof(XRBaseInteractor).GetField("m_AttachTransform", BindingFlags.Instance | BindingFlags.NonPublic);
+
+            if (attachField != null)
+            {
+                attachField.SetValue(SnapZone, attach.transform);
+            }
+            else
+            {
+                DestroyImmediate(attach);
+
+                // It seems that the Unity team has renamed that field. Check the SnapZone script. 
+                Debug.Log("We failed to add an attach point for this snap zone property. Please, drop us a word if you see this message.");
+            }
+
+            List<Collider> colliders = GetComponentsInChildren<Collider>().Where(foundCollider => foundCollider.isTrigger).ToList();
+
+            if (colliders.Count == 0)
+            {
+                Debug.Log($"{addAttachAtOrigin} {tooLittleCollidersError} {attachPointExplanation}");
+                Debug.LogWarning("This snap zone has no trigger colliders. It will not snap any objects, and will physically block them. Add a collider to the snap zone object if it has none. Check the Is Triggered box of its collider.");
```

Comment:
Why would physically block anything if it might not have colliders?
Also, this sounds odd: `Add a collider to the snap zone object if it has none.` since we kind of already stated that there are not valid colliders.